### PR TITLE
Update functionMap.php for SplFileInfo::getPathInfo return type

### PR DESCRIPTION
**Pull request by DannyvdSluijs**
SplFileInfo::getPathInfo can return null as per documentation https://www.php.net/manual/en/splfileinfo.getpathinfo.php

After trying to report the issue in the distribution repo I was sent here, see https://github.com/phpstan/phpstan/issues/new?assignees=&labels=&projects=&template=Function_signature_mismatch.md. So I hope this is the right place and the right changes. Feel free to ping me if anything else is needed.

This mismatch was discovered using the following [code snippet](https://phpstan.org/r/82ae7b7b-aa9a-4dbc-9f08-146614fd4570)
```php
<?php declare(strict_types = 1);


$file = tempnam(sys_get_temp_dir(), 'phpstan-');
$info = new \SplFileInfo($file);

echo $info->getPathInfo()?->getPathname();
```

### CODE REVIEW

**Comment by DannyvdSluijs** [comment]
> You've opened the pull request against the latest branch 2.0.x. PHPStan 2.0 is not going to be released for months. If your code is relevant on 1.12.x and you want it to be released sooner, please rebase your pull request and change its target to 1.12.x.

Rebased and retargetted to `1.12.x`

File: resources/functionMap.php [hunk 1]
```diff
@@ -11556,7 +11556,7 @@
 'SplFileInfo::getMTime' => ['__benevolent<int|false>'],
 'SplFileInfo::getOwner' => ['__benevolent<int|false>'],
 'SplFileInfo::getPath' => ['string'],
-'SplFileInfo::getPathInfo' => ['SplFileInfo', 'class_name='=>'string'],
+'SplFileInfo::getPathInfo' => ['SplFileInfo|null', 'class_name='=>'string'],
```
Comment (ondrejmirtes) [inline resources/functionMap.php]:
```suggestion
'SplFileInfo::getPathInfo' => ['__benevolent<SplFileInfo|null>', 'class_name='=>'string'],
```

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
I'd rather be a little bit calmer if it was a benevolent union type, so that it doesn't bother people who are sure it cannot fail for them.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**